[hannahtay] iP

hannahtay opened the pull request.

hannahtay committed:
Added Level 0 - Rename, greet, exit

hannahtay committed:
Added echo function, new class for system messages

Echoes user inputs
Made code neater

hannahtay committed:
Added list function

Users now able to add items to the todo list
When input "list", all previously added items appear in order

hannahtay committed:
Added unmark and mark done commands

hannahtay committed:
Added tracking for ToDos, events, deadlines

able to view type of task from 'list'
able to add deadlines & event from/to commands

Li-JunXian commented on src/main/java/sysMsgs.java:
```diff
@@ -0,0 +1,18 @@
+public class sysMsgs {
+
+    public void showIntro() {
+        String logo =
+                "     /ᐠ,,,,∧\n"
+                        + "     (֊⎚-⎚)\n"
+                        + "     /⌒   づ\uD83D\uDCD7";
+        System.out.println(logo);
+        System.out.println("Hi there! I'm NYANI ^_^");
+        System.out.println("What shall we do today?");
+    }
```
Great compartmentalisation! Separation between the greeting introductory message and the exit message makes it clear for future review and amendments (if required).

But since the name of the class is "sysMsgs", perhaps it would be better to name the method as a noun corresponding to your naming convention for the "exitMsg".

Li-JunXian commented on src/main/java/todoList.java:
```diff
@@ -0,0 +1,126 @@
+
+    public void showTasks() {
+        if (tasks.isEmpty()){
+            System.out.println("──────────୨ৎ──────────");
```
I like the artistic horizontal line, perhaps maybe it can also be assigned to a variable for easier future use.

Li-JunXian commented on src/main/java/sysMsgs.java:
```diff
@@ -0,0 +1,18 @@
+public class sysMsgs {
```
Is there a specific reason to separate the "sysMsgs" class from "Nyani" class?

Li-JunXian commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,65 @@
+                    if (toPart.length == 2) {
+                        toDo.addTask(new Event(fromPart[0], toPart[0], toPart[1]));
+                    } else {
+                        System.out.println("wrong format! try event /from [date/time] /to [date/time]!");
+                    }
+                } else {
+                    System.out.println("wrong format! try event /from [date/time] /to [date/time]!");
```
I like how you really specify every detail for users to fully understand what they should do, great job! Perhaps try to combine these 2 lines as one since they output the same message ;)

Li-JunXian commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,65 @@
+                if (parts.length == 2) {
+                    toDo.addTask(new Deadline(parts[0], parts[1]));
+                } else {
+                    System.out.println("wrong format! try deadline [task] /by [date/time]!");
```
Great attention to all details, maybe we can try saving this sentence template as an assignment for more convenient future implementation :)

Li-JunXian reviewed (commented):
Good job, great attentions to details and user experience ;~)

J030104 commented on src/main/java/todoList.java:
```diff
@@ -0,0 +1,126 @@
+import java.util.List;
+import java.util.Scanner;
+
+abstract class Task {
```
Well organized : )

J030104 commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,65 @@
+                }
+            }
+
+            else if (input.startsWith ("deadline ")) {
```
I guess you might want to make your coding style consistent. There's a blank after startWith in this line but not all of the function call are like this.

J030104 commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,65 @@
+        systemOut.showIntro();
+        todoList toDo = new todoList();
+
+        while (true) {
```
If possible, try integrating the echo function into your current code so that your bot is more powerful.

J030104 commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,65 @@
+
+        while (true) {
+            String input = sc.nextLine();
+
```
I myself would add a help function here (like giving instructions on how user can interact with the bot) (on startup).

hannahtay committed:
Added command 'help', tidied up code

'help': lists out all commands user can call

hannahtay committed:
Added exception class and exception messages

hannahtay committed:
Revert "Added exception class and exception messages"

This reverts commit 72036d7c92d0f7060776acc3f66ecf12a8183558.

hannahtay committed:
Added exceptions and exception messages

hannahtay committed:
Added exception class and edited exception messages

hannahtay committed:
branch-Level-5

Merging level 5 increment into master

hannahtay committed:
Added exception class and error messages

hannahtay committed:
Merge branch 'branch-Level-5'

hannahtay committed:
Deleted extra file caused by technical errors

hannahtay committed:
Added delete task function

hannahtay committed:
Added data saver

Items can be saved into a Nyani.txt file.
However, the saved data does not show up upon typing the 'list' command on 2nd run

hannahtay committed:
Edited task message upon load

hannahtay committed:
Shifted adding/marking system messages to main

hannahtay committed:
Edited system message

hannahtay committed:
Added loading of data function

Can finally load data in files to the list

hannahtay committed:
Merge branch 'branch-Level-7'

hannahtay committed:
Edited cat logo

hannahtay committed:
Edited problems with printing deadline

hannahtay committed:
Changed special symbols that does not appear in terminal

hannahtay committed:
Rearranged code to fit OOP

added parser, command classes to improve readability of code

hannahtay committed:
Added find command

hannahtay committed:
Added java doc comments

hannahtay committed:
Merged branch 'branch-A-JavaDoc'

hannahtay committed:
Added missing code from merging

hannahtay committed:
Updated README

hannahtay committed:
Updated UG 

Updated for todo and deadline

hannahtay committed:
Updated UG 

Finished remaining features

hannahtay committed:
Updated find command

was previously missing after merge

hannahtay committed:
Updated UG

hannahtay committed:
Merging branch 'master'

hannahtay committed:
Merging updated UG

hannahtay committed:
Updated UG

hannahtay committed:
Made the saving compatible with all devices

Saved txt file to computer under 'Documents', not all may have this directory.

Change file path.

Changing file path to one that is directly under the repo itself confirms usage and compatibility across devices.

hannahtay committed:
Update UG

hannahtay committed:
Update UG

hannahtay committed:
Update UG

hannahtay committed:
Edit exception message for unavailable commands

hannahtay committed:
Update UG

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,185 @@
+ * Each command must implement the execute method and specify whether it is an exit command.
+ */
+
+abstract class Command {
```
any specific reason not including the access specifier?

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,185 @@
+/**
+ * Represents the command to list all tasks in the to-do list.
+ */
+class ListCommand extends Command {
```
better to write classes in their individual files

okkhoy commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,185 @@
+    public boolean isExit() {
+        return false;
+    }
+}
```
you can group the classes using packages; that may be better than writing all related classes in the same file

okkhoy commented on src/main/java/Nyani.java:
```diff
@@ -0,0 +1,38 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by FernFlower decompiler)
+//
```
errrm... why?

okkhoy commented on src/main/java/SaveDataHandler.java:
```diff
@@ -0,0 +1,121 @@
+            return;
+        }
+
+        // Write tasks to specified file
```
redundant comment

okkhoy commented on src/main/java/SaveDataHandler.java:
```diff
@@ -0,0 +1,121 @@
+                    System.out.println(line);
+
+                    // Split the line into status and the rest of the info
+                    String[] parts = line.split("\\]", 2);
```
consider moving low-level string operations to a different method for better SLAP

okkhoy commented on src/main/java/todoList.java:
```diff
@@ -0,0 +1,223 @@
+/**
+ * ToDo class representing a task classified under 'to-do'.
+ */
+class ToDo extends Task {
```
same comment as Command class file

okkhoy reviewed (commented):
@hannahtay 
please consider reorganizing the code with packages. see comments for details

hannahtay committed:
Refactor code to separate extended classes from main class

Edited the command and todoList classes

hannahtay committed:
Add packages

hannahtay committed:
Merge pull request #2 from hannahtay/branch-packages

Add packages

hannahtay committed:
Delete data file

hannahtay committed:
Update README.md

hannahtay committed:
Update README.md

hannahtay committed:
Update README.md
